packages/discovery-provider/src/queries/get_comments.py

```diff
@@ -78,35 +73,37 @@ def get_track_comments(args, track_id):
 
     with db.scoped_session() as session:
         track_comments = (
-            session.query(Comment)
+            session.query(Comment, func.count(CommentReaction.comment_id).label("react_count"))
             .outerjoin(
                 CommentThreadAlias,
                 Comment.comment_id == CommentThreadAlias.comment_id,
             )
+            .outerjoin(CommentReaction, Comment.comment_id == CommentReaction.comment_id)
+            .group_by(Comment.comment_id)
             .filter(
                 Comment.entity_id == track_id,
                 Comment.entity_type == "Track",
                 CommentThreadAlias.parent_comment_id
                 == None,  # Check if parent_comment_id is null
                 Comment.is_delete == False,
             )
-            .order_by(asc(Comment.created_at))
+            .order_by(order_by)
             .offset(offset)
             .limit(limit)
             .all()
         )
 
         return [
             {
-                "id": encode_int_id(track_comment.comment_id),
-                "user_id": track_comment.user_id,
-                "message": track_comment.text,
-                "is_pinned": track_comment.is_pinned,
-                "track_timestamp_s": track_comment.track_timestamp_s,
-                "react_count": get_reaction_count(session, track_comment.comment_id),
-                "replies": get_replies(session, track_comment.comment_id),
-                "created_at": str(track_comment.created_at),
-                "updated_at": str(track_comment.updated_at),
+                "id": encode_int_id(track_comment[0].comment_id),
```

Comment:
not in love with this btw.

Reply:
hmm interesting, so your query above turns this into an array? I'm not familiar enough with this logic to know if theres a better way to do that. 
BUT I do believe there's a way in python to destructure to avoid the `[0]` and `[1]` everywhere

(don't quote me on this but I think its just this)
```python
for track_comment, react_count in track_comments
```